lighthouse/classes/event_properties/interfaces.py

```diff
@@ -0,0 +1,225 @@
+    inside this class.
+    """
+
+    @abstractmethod
+    def validate(self) -> bool:
+        """
+        Retuns a boolean (True or False) indicating if the params received for this
+        EventProperty are correct in order to retrieve the data. If there is an error
+        in validation it will add it to the errors list.
+        This methods is safe and should not raise any exception.
+
+        Arguments:
+            None
+
+        Returns:
+            bool - True/False depending on the result of the validation.
+
+        """
+        ...
+
+    @abstractmethod
+    def errors(self) -> List[str]:
+        """
+        Returns the complete list of errors found.
+
+        Arguments:
+            None
+
+        Returns:
+            List[str] - List of error messages found currently
+        """
+        ...
+
+    @abstractmethod
+    def value(self) -> Any:
+        """
+        Returns the value for the property. If the value cannot be obtained or
+        does not have a valid value, then it raises an exception.
+        NB: To avoid this it, the instance should be checked first with the
+        validate() method.
+
+        Arguments:
+            None
+
+        Returns:
+            Any - Value of the property
+
+        """
+        ...
+
+    @abstractmethod
+    def add_to_warehouse_message(self, message: WarehouseMessage) -> None:
+        """
+        Adds this event property information into the warehouse message
+
+        Arguments:
+            message: WarehouseMessage - A building message where we want to write
+            the information from this event property
+
+        Returns:
+            None
+
+        """
+        ...
+
+
+class EventPropertyAbstract(EventPropertyInterface):
+    """
+    This class provide some tools to validate data obtained directly
+    from the params argument.
+    It is an abstract class that will need to define the methods specified
+    in its interface class (EventPropertyInterface).
+    If a more advanced process is needed (like accessing an external service),
+    then the methods of this class should be rewritten (or a new class
+    should be created).
+    This class also contains methods that will be needed for internal
+    process for the class but should not be called from outside of this
+    instance (they should be considered all protected methods).
+    The public interface that should be used in this case is
+    defined in EventPropertyInterface.
+    """
+
+    def __init__(self, params: Dict[str, Any]):
+        """
+        Stores the params, resets the instance and perform a validation check on the params
+
+        Arguments:
+            params: Dict[str, Any] - dict of key/value with the settings for
+            this property
+        """
+        self.reset()
+        self._params = params
+        if self._params is None:
+            raise ValidationError("You need to define params to create the EventProperty")
+
+    def reset(self):
+        """
```

Comment:
Is only to be use in the initialization. There are a group of event properties that are "Simple" and have this signature:

``` def __init__(self, params: Dict[str, Any])```

but there are other properties that are "Complex" and have this:

```def __init__(self, cherrytrack_wells_from_destination: CherrytrackWellsFromDestination):```

However in both I want to use what is defined in reset(), but I didnt know how to run the contents of reset in the init by calling super().__init__() because the arguments are going to be different. Having reset() was one way of doing it.